src/connector/src/sink/file_sink/opendal_sink.rs

```diff
@@ -178,43 +196,67 @@ pub struct OpenDalSinkWriter {
+    /// successful, it returns the last `chunk_id` that was written;
+    /// otherwise, it returns None.
+    ///
+    ///
+    /// This method is currently intended for use by the file sink's writer.
+    pub async fn write_batch_and_try_finish(
+        &mut self,
+        chunk: StreamChunk,
+        chunk_id: usize,
+    ) -> Result<Option<usize>> {
         if self.sink_writer.is_none() {
-            self.create_sink_writer(epoch).await?;
-        }
-        if self.is_append_only {
-            self.append_only(chunk).await
+            self.create_sink_writer().await?;
+        };
+
+        // While writing this chunk (via `append_only`), it checks if the maximum row count
+        // has been reached to determine if the file should be finalized.
+        // Then, it checks the rollover interval to see if the time-based batching threshold
+        // has been met. If either condition results in a successful write, the write is marked as complete.
+        let finish_write =
+            self.append_only(chunk).await? || self.try_finish_write_via_rollover_interval().await?;
+        self.written_chunk_id = Some(chunk_id);
+        if finish_write {
+            Ok(Some(chunk_id))
         } else {
-            // currently file sink only supports append only mode.
-            unimplemented!()
+            Ok(None)
         }
     }
 
-    async fn begin_epoch(&mut self, epoch: u64) -> Result<()> {
-        self.epoch = Some(epoch);
-        Ok(())
-    }
-
+    /// Attempts to complete the file write using the writer's internal batching strategy.
+    ///
+    /// If the write is completed, returns the last `chunk_id` that was written;
+    /// otherwise, returns None.
+    ///
     /// For the file sink, currently, the sink decoupling feature is not enabled.
     /// When a checkpoint arrives, the force commit is performed to write the data to the file.
     /// In the future if flush and checkpoint is decoupled, we should enable sink decouple accordingly.
-    async fn barrier(&mut self, is_checkpoint: bool) -> Result<()> {
-        if is_checkpoint && let Some(sink_writer) = self.sink_writer.take() {
-            match sink_writer {
-                FileWriterEnum::ParquetFileWriter(w) => {
-                    let _ = w.close().await?;
-                }
-            };
+    pub async fn try_finish(&mut self) -> Result<Option<usize>> {
+        match self.try_finish_write_via_rollover_interval().await? {
+            true => Ok(self.written_chunk_id),
+            false => Ok(None),
         }
+    }
+}
+#[async_trait]
+impl SinkWriter for OpenDalSinkWriter {
+    async fn write_batch(&mut self, _chunk: StreamChunk) -> Result<()> {
+        Ok(())
+    }
+
+    async fn begin_epoch(&mut self, epoch: u64) -> Result<()> {
+        self.epoch = Some(epoch);
+        Ok(())
+    }
 
+    async fn barrier(&mut self, _is_checkpoint: bool) -> Result<Self::CommitMetadata> {
         Ok(())
     }
 }
```

Comment:
We need to implement `begin_epoch` function，but think twice this may be useless. As we do not use epoch in `OpendalSinkWriter` , maybe we don’t even need to store epochs in the writer. IIRC previously epoch is only used in file naming. Pls correct me is I'm wrong.